src/db.rs

```diff
@@ -103,20 +103,30 @@ impl DbInner {
         self.check_error()?;
         let snapshot = self.state.read().snapshot();
 
+        macro_rules! as_result {
+            ($deletable:expr) => {
+                match $deletable {
+                    ValueDeletable::Value(v) => Ok(Some(v)),
+                    ValueDeletable::Merge(_) => Err(SlateDBError::MergeUnsupported),
+                    ValueDeletable::Tombstone => Ok(None),
+                }
+            };
+        }
```

Comment:
i guess this could be implemented as a method of `ValueDeletable` instead of a macro? 

like:

```
impl TryInto<Option<Bytes>> for ValueDeletable {
  type Error = SlateDBErrror;
  ...
}
```

Reply:
That's more or less reintroducing the method I had a hard time getting rid of in one of the commits (`feat(MergeOperator): Remove ValueDeletable#into_optional.`).

I have a strong preference to not have such method at all, because it can often lead to wrong assumptions that `Value` & `Merge` can be treated equally.

The macro is temporary (to avoid boilerplate) and this whole method will be reworked in one of the next PRs, where I need to introduce actual handling for ValueDeletable#Merge.

Does that sound ok?

Reply:
i personally have some bias against using macros, following [The Principle of Least Power](https://blog.codinghorror.com/the-principle-of-least-power/). when it's possible to use methods or even boilerplate code, it's better to consider macros as the last resort.

if this macro is planned to be reworked in future PRs, and we want to avoid the previous error-prone methods like `into_optional`, I think keeping the boilerplate code is fine 🤔.

Reply:
> i personally have some bias against using macros, ...

In C/C++ world would I'd strongly agree, but in rust it feels way safer and more robust.

Anyway, it seems we can actually have a temporary local / inline implementation of TryInto, which 'm fine with. -> https://github.com/slatedb/slatedb/pull/336/commits/6c08fb75b7d8e8f9ca32d154f4bc5c8d0fa13f45

Reply:
I like it 👍